homeassistant/components/ozw/services.py

```diff
@@ -37,6 +43,98 @@ def async_register(self):
             ),
         )
 
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
+            self.async_set_config_parameter,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CONFIG_PARAMETER): vol.Coerce(int),
+                    vol.Required(const.ATTR_CONFIG_VALUE): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                }
+            ),
+        )
+
+    @callback
+    def async_set_config_parameter(self, service):
+        """Set a config parameter to a node."""
+        instance_id = service.data[const.ATTR_INSTANCE_ID]
+        node_id = service.data[const.ATTR_NODE_ID]
+        param = service.data[const.ATTR_CONFIG_PARAMETER]
+        selection = service.data[const.ATTR_CONFIG_VALUE]
+        payload = None
+
+        node = self._manager.get_instance(instance_id).get_node(node_id).values()
+
+        for value in node:
+            if (
+                value.command_class != CommandClass.CONFIGURATION
+                or value.index != param
+            ):
+                continue
+
+            if value.type == ValueType.BOOL:
+                payload = selection == "True"
+
+            if value.type == ValueType.LIST:
+                # accept either string from the list value OR the int value
+                if isinstance(selection, int):
+                    if selection > value.max or selection < value.min:
+                        _LOGGER.error(
+                            "Value %s out of range for parameter %s (Min: %s Max: %s)",
+                            selection,
+                            param,
+                            value.min,
+                            value.max,
+                        )
+                        break
+                    payload = int(selection)
+
+                # iterate list labels to get value
+                for selected in value.value["List"]:
+                    if selected["Label"] != selection:
+                        continue
+                    payload = int(selected["Value"])
+
+                if payload is None:
+                    _LOGGER.error(
+                        "Invalid value %s for parameter %s", selection, param,
+                    )
+                    break
+
+            if value.type == ValueType.BUTTON:
+                # Unsupported at this time
+                _LOGGER.info("Button type not supported yet")
+                return
```

Comment:
I think that we should be consistent, and either use `break` or use `return`  to exit the loop

Reply:
ill get them changed to a single type

Reply:
fixed in latest commit